notebooks/analysis/notebooks/Actions.ipynb

```diff
@@ -0,0 +1,156 @@
+   "execution_count": null,
+   "id": "8b240d92-24a9-48a4-a750-f3f111dfa212",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import pandas as pd\n",
+    "\n",
+    "from weco_datascience.reporting import get_recent_data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "34681ded-d95e-4c35-b859-40f1be32e3fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = get_recent_data(config=os.environ, n=10000, index=\"metrics-conversion-prod\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b48f9fb0-e3d8-4773-a5e8-37662e20be4e",
+   "metadata": {},
+   "source": [
+    "## [Definitions of actions] (https://app.gitbook.com/o/-LumfFcEMKx4gYXKAZTQ/s/MM8zK9GZyFCpy8kvd5Km/analytics-and-reporting/kibana-definitions) are translated into code below ##"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "604ac0df-ea66-4f17-91ac-c30a4a729396",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "concepts = df.loc[(df[\"page.name\"]==\"concepts\")]\n",
+    "concepts[\"action\"]=\"concepts\"\n",
+    "search = df.loc[(df[\"page.name\"]==\"works\") & (df[\"page.query.query\"]!=\" \")]\n",
+    "search[\"action\"]=\"search\"\n",
+    "works =  df.loc[(df[\"page.name\"]==\"work\") & (df[\"page.query.workId\"]!=\" \")]\n",
+    "works[\"action\"]=\"works\"\n",
+    "item_view = df.loc[(df[\"page.name\"]==\"item\") & (\n",
+    "    df[\"properties.event\"]!=\"download\") & (df[\"source\"]!=\"viewer/scroll\")]\n",
+    "item_view[\"action\"]=\"item_vew\"\n",
+    "image_view =  df.loc[(df[\"page.name\"]==\"image\") & (df[\"type\"]==\"pageview\")]\n",
+    "image_view[\"action\"]=\"image_view\"\n",
+    "image_modal = df.loc[(df[\"page.name\"]==\"images\") & (\n",
+    "    df[\"properties.event\"]==\"Open image modal\") & (df[\"page.query.query\"]!=\" \")]\n",
+    "image_modal[\"action\"]=\"image_modal\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a7a55b02-550f-42fe-87fa-4800a90c6767",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "selected=[concepts, search, works, item_view, image_view, image_modal]\n",
+    "df2=pd.concat(selected)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2000d44e-0222-43e3-af51-80da39d19ab0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "keepers=[\"@timestamp\", \"anonymousId\", \"page.query.query\", \"page.query.workId\", \"page.name\", \"properties.event\", \"action\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fb9645ac-12a6-4b5a-b77f-8e6949d22df5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df2[(keepers)]=df2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "610b47db-3c7c-4c0c-90f1-504128b2a4a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Number of actions:\", len(df2))"
```

Comment:
Do these results answer the questions you set out to investigate? I don't know whether the data's being analysed in more detail somewhere else, but if so, it might be good to include a link to that resource.

Reply:
No, this code is preliminary to me shoveling it into a decision tree format.

Reply:
Nice! As I said below, I think it would be great to include that context in the notebook's introduction

Reply:
Actually, I had a go at this last night and decided a tree map is probably sufficient for a first stab.